File: app/src/main/java/com/testeandroidv2/loginScreen/LoginConfigurator.java
package com.testeandroidv2.loginScreen;

import java.lang.ref.WeakReference;


public enum LoginConfigurator {
    INSTANCE;

    public void configure(LoginActivity activity) {

        LoginRouter router = new LoginRouter();
        router.activity = new WeakReference<>(activity);

        LoginPresenter presenter = new LoginPresenter();
        presenter.output = new WeakReference<LoginActivityInput>(activity);

        LoginInteractor interactor = new LoginInteractor();
        interactor.output = presenter;

        if (activity.output == null) {
            activity.output = interactor;
        }
        if (activity.router == null) {
            activity.router = router;
        }
    }
}

File: app/src/main/java/br/com/cauejannini/testesantander/login/LoginConfigurator.kt
package br.com.cauejannini.testesantander.login

import java.lang.ref.WeakReference

object LoginConfigurator {

    fun configure(loginActivity: LoginActivity) {

        val presenter = LoginPresenter()
        presenter.output = WeakReference(loginActivity)

        val interactor = LoginInteractor()
        interactor.output = presenter

        loginActivity.output = interactor

    }

}